scripts/code-check.sh

```diff
@@ -2,9 +2,16 @@
 
 # Run some pre commit checks on the Go source code. Prevent the commit if any errors are found
 
-# Format the Go code
+# Check if the Go code is formatted
 check_code_format (){
-    go fmt $(go list ./... | grep -v /vendor/)
+    gofmt -l . | grep -v vendor/ >/dev/null 2>&1
+    if [ $? -ne 1 ]; then
```

Comment:
Will this approach still format the code for the user, so that they can just stage the files and commit?

Reply:
It will check if the user would like to format the code them is required to use the command "make fmt" for example. Would you like that both happens?

Reply:
I think it's better to ask the user to run the command. I had some problem with files getting changed locally when I was trying to interactively rebase to update some commits and I kept getting the additional files added there without me confirming it. The files that were being synced had no relation to my changes as they were in the go server and I was working on the ui. I didn't make sense to include them in that pr.

Reply:
I agree with you @laurafitzgerald sametimes the magic make the things more complicated than easier and make more sense shows the error and not fix without the user knows what and why.